src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -1128,6 +1128,147 @@ void MacroAssembler::wrap_label(Register r1, Register r2, Label &L,
 
 #undef INSN
 
+// cmov
+void MacroAssembler::cmov_eq(Register cmp1, Register cmp2, Register dst, Register src) {
+  if (UseZicond) {
+    xorr(t0, cmp1, cmp2);
+    czero_eqz(dst, dst, t0);
+    czero_nez(t0 , src, t0);
+    orr(dst, dst, t0);
+    return;
+  }
+  Label no_set;
+  bne(cmp1, cmp2, no_set);
+  mv(dst, src);
+  bind(no_set);
```

Comment:
Do we have performance data comparison between `-UseZicond ` and `+ UseZicond `?

Reply:
Seems the critical path when `+UseZicond` is longer than `-UseZicond`, I'm not sure how much impact the branch code will bring when `-UseZicond`.

Reply:
Yes, exactly, a branch is always more 'intrusive' in the pipeline than plain scalar ops.
As we discussed, there are cases where czero perform in pair with branch, but in general it's a win.